grid.cpp

```diff
@@ -266,7 +266,9 @@ void initializeGrids(
    // Update technicalGrid
    technicalGrid.updateGhostCells(); // This needs to be done at some point
 
-   if (!P::isRestart) {
+   if (!P::isRestart && !P::writeFullBGB) {
```

Comment:
I can't very easily brain this triple-negative with an exception. Suggestion for a new comment:

If we are starting a new regular simulation, we need to prepare all cells with their initial state.